File: cpp/1679.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int i=0;
        int j=nums.size()-1;    
        int count = 0;
        while(i<j) {
            if(nums[i]+nums[j]<k) {
                i++; 
                continue;
            }
            else if(nums[i]+nums[j]>k) {
                j--;
                continue;
            }
            count++;
            i++;
            j--;
        }
        return count;
    }
};

File: python/1679.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        d = dict()
        for i in nums:
            if(i not in d.keys()):
                d[i] = 1
            else:
                d[i] += 1

        res = 0
        for i in d:
            if(d[i] == 0):
                continue
            if(i == k-i):
                res+=d[i]//2
            elif(k-i in d):
                res+=min(d[k-i],d[i])
                d[k-i] = 0

        return res